VMs/metrics/README.md

```diff
@@ -0,0 +1,243 @@
+
+Our configuration is not perfect (the shortest time we can set to kill off an inactive
+node is one minute). Because of this, if another job is in the queue or arrives within
+one minute of a previous job completing, it is scheduled to build on the same VM instance.
+The expectation is that we have enough gaps that we acquire sufficient new VMs to vastly
+reduce the 'dirty node' instabilities we see with pure bare-metal builds.
+
+In the event of getting a dirty VM node, executing a 'node offline/online' in the
+Jenkins UI runs a VM shutdown/clone/run cycle. This gets the system back to having
+a stable, clean VM on that node.
+
+The following are examples of the configuration dialogs:
+
+### Setup the pre-post scripts
+
+On your slave VM node host machine, create a pair of scripts specific to your host to allow
+Jenkins to invoke the sub-scripts. For example:
+
+`start_gitvm.sh`
+```bash
+#!/bin/bash
+ROOTDIR=${HOME}/kata-containers-ci/VMs/metrics
+
+if [ $# -ne 1 ]; then
+	echo "Require VM config name as only parameter"
+	exit 1
+fi
+
+cd ${ROOTDIR}
+./6_start_clone_agent.sh $1
+```
+
+`stop_gitvm.sh`
+```bash
+#!/bin/bash
+ROOTDIR=${HOME}/kata-containers-ci/VMs/metrics
+
+if [ $# -ne 1 ]; then
+	echo "Require VM config name as only parameter"
+	exit 1
+fi
+
+cd ${ROOTDIR}
+./7_delete_clone.sh $1
+```
+
+Add those scripts to the Jenkins node agent launch dialog, along with an `echo` hack
+to prevent Jenkins from launching the agent jar itself, such as the following:
+
+![prefix and suffix](pictures/prefix_suffix.png)
+
+The passing of the `config_dir` parameter to the scripts from the Jenkins node
+config dialog. This allows us to set up separate nodes for different configurations
+to test.
+
+It is recommended that you only schedule a single VM build on each bare metal host
+at a time. This can be achieved through use of labels, resources, and locks in the
+Jenkins UI to enforce exclusivity and serialisation.
+
+### Set up the short lived nodes
+
+To try and get a clean VM clone for each PR, we set the node to take itself offline
+as soon as it is idle, which shuts down and removes the previous cloned VM.
+
+![Offline when idle](pictures/offline_idle.png)
+
+Jenkins does not currently allow us to configure the Idle delay to less than one minute.
+
+## Node specific modifications
+
+If you need to make node specific modifications such as:
+
+- Modify the checkmetrics toml file to take into account the performance of your node.
+- Add in site specific configs, such as PROXY settings.
+- Modify paths, such as the `Jenkins` user HOME path or path to the scripts.
+
+The recommended procedure for any of the previous modifications is to make a fork of
+this repository and keep your local modifications in a branch that tracks this main
+repository.
+
+### Proxies
+
+To configure proxies in the cloud-init `user-data` file is complex, as we need
+to set up the proxies for a number of programs, including, `apt`, `git` and `docker`. To
```

Comment:
Adding in the series comma after "git": to set up the proxies for a number of programs, including, `apt`, `git`, and `docker`.

Reply:
done